tests/test_casereader.py

```diff
@@ -202,7 +202,7 @@ def test_case_reader_get_rp_and_config_vars():
     assert reader.rp_var.pressure.output_dpdt__q() is True
     assert len(reader.rp_var.context.map_r17__plus()) == 53
     assert reader.rp_var.defaults.pre_r19__dot0_early__q() is False
-    with pytest.raises(BaseException):
+    with pytest.raises(Exception):
```

Comment:
generally better to test the specific exception which is thrown in this case

Reply:
Not only in tests, but ideally all of the exception catches should be as specific as possible

Regardless, not criticizing as this is still a big improvement, thanks @prmukherj

Reply:
Done, Thank you @mkundu1.

Thank you @raph-luc, I totally agree with the point you raised. However, covering all the specific exceptions for the cases where these changes have been made seems a bit tricky. That's why for now I think we can re-raise the Exception using the from keyword which provides user the information about the original error as well. Seems fine for now, right?

Reply:
That is right, I fully agree